ddl/ddl.go

```diff
@@ -681,6 +684,44 @@ func (d *ddl) Start(ctxPool *pools.ResourcePool) error {
 	return nil
 }
 
+// EnableDDL enable this node to execute ddl.
+// Since ownerManager.CampaignOwner will start a new goroutine to run ownerManager.campaignLoop,
+// we should make sure that before invoking EnableDDL(), ddl is DISABLE.
+func (d *ddl) EnableDDL() error {
+	err := d.ownerManager.CampaignOwner()
+	return errors.Trace(err)
+}
+
+// DisableDDL disable this node to execute ddl.
+// We should make sure that before invoking DisableDDL(), ddl is ENABLE.
+func (d *ddl) DisableDDL() error {
+	if d.ownerManager.IsOwner() {
+		// If there is only one node, we should NOT disable ddl.
+		serverInfo, err := infosync.GetAllServerInfo(d.ctx)
+		if err != nil {
+			logutil.BgLogger().Error("[ddl] error when GetAllServerInfo", zap.Error(err))
+			return err
+		}
+		if len(serverInfo) <= 1 {
+			return dbterror.ErrDDLSetting.GenWithStackByArgs("can not disable ddl when there is only one instance")
+		}
+
+		// FIXME: if possible, when this node is the only node with DDL, ths setting of DisableDDL should fail.
+
+		// lets the owner start a new election.
+		err = d.ownerManager.ResignOwner(d.ctx)
+
+		if err != nil {
+			logutil.BgLogger().Error("[ddl] error when ResignOwner", zap.Error(err))
+			return err
+		}
```

Comment:
We also passed the parameter `ctx` to `watchOwner`. Could we remove this logic？

Reply:
Fixed it. Now a `ctx` of `Manager` itself is passed in.

Reply:
That's not what I meant. The `CampaignCancel()` is executed. We passed `ctx` in `Campaign` and `watchOwner`, so I think we will quit the owner campaign. So we don't need those lines of logic?